src/LinAlg/hiopVectorRajaImpl.hpp

```diff
@@ -2062,48 +2108,43 @@ void hiopVectorRajaPar::print(FILE* file/*=nullptr*/, const char* msg/*=nullptr*
     }    
     fprintf(file, "=[");
     max_elems = max_elems >= 0 ? max_elems : n_local_;
-    for(int it=0; it<max_elems; it++) {
+    for(int it=0; it<max_elems; it++)
       fprintf(file, "%22.16e ; ", data_host_[it]);
-    }
     fprintf(file, "];\n");
   }
 }
 
-void hiopVectorRajaPar::copyToDev()
+template<class MEM, class POL>
+void hiopVectorRaja<MEM,POL>::print() const
 {
-  if(data_dev_ == data_host_)
-    return;
-  auto& resmgr = umpire::ResourceManager::getInstance();
-  resmgr.copy(data_dev_, data_host_);
-}
-
-void hiopVectorRajaPar::copyFromDev()
-{
-  if(data_dev_ == data_host_)
-    return;
-  auto& resmgr = umpire::ResourceManager::getInstance();
-  resmgr.copy(data_host_, data_dev_);
+  auto* This = const_cast<hiopVectorRaja<MEM, POL>* >(this);
+  assert(nullptr != This);
+  This->copyFromDev();
+  for(index_type it=0; it<n_local_; it++) {
+    printf("vec [%d] = %1.16e\n",it+1,data_host_[it]);
+  }
+  printf("\n");
```

Comment:
```suggestion
  auto* temp_vec = const_cast<hiopVectorRaja<MEM, POL>* >(this);
  assert(nullptr != temp_vec);
  temp_vec->copyFromDev();
  for(index_type it=0; it<n_local_; it++) {
    printf("vec [%d] = %1.16e\n",it+1,data_host_[it]);
  }
  printf("\n");
```
Perhaps this is too pedantic...

Reply:
+1 Variable name `This` is too close to keyword `this`.

Reply:
fixed